sys/include/ztimer.h

```diff
@@ -460,6 +460,22 @@ static inline ztimer_now_t ztimer_now(ztimer_clock_t *clock)
     }
 }
 
+/**
+ * @brief   Get the time until a timer will trigger
+ *
+ * @param[in]   clock          ztimer clock to operate on
+ * @param[in]   timer          ztimer to messuers to
+ *
+ * @return  Current time until the @p timer is trigger in clock units
+ */
+static inline uint32_t ztimer_until(ztimer_clock_t *clock, ztimer_t *timer){
+#if ZTIMER_UNTIL_SAFTY_NET
+    if (!ztimer_is_set(clock,timer)){ return 0; }
+#endif
+    return timer->base.offset - (uint32_t) ztimer_now(clock);
```

Comment:
this doesn't work:
- the base.offset is relative to clock.list.offset and all preceding timers in the list
- as is, this is racey, e.g., this fails horribly if the timer passes after the ztimer_is_set()

Reply:
>     * the base.offset is relative to clock.list.offset and all preceding timers in the list
? Why ?

Reply:
https://github.com/RIOT-OS/RIOT/blob/d5734013197cbdca692a3852736dfa6720f74697/sys/include/ztimer.h#L112

- the list is storing 32bit relative values in order to not not use absolute target times
- there needs to be some absolute reference value, so list.base.offset is used
- the offsets are relative to the last entry, so updating the head offset doesn't require updating all timer offsets, just the one that's added and possibly the following.